src/app/client/src/app/modules/shared-feature/components/validate-teacher-identifier-popup/validate-teacher-identifier-popup.component.ts

```diff
@@ -1,47 +1,112 @@
+import { IInteractEventObject, IInteractEventEdata } from '@sunbird/telemetry';
 import { ResourceService } from '@sunbird/shared';
-import { Component, OnInit, ViewChild, ElementRef, Output,  EventEmitter  } from '@angular/core';
+import { Component, OnInit, ViewChild, ElementRef, Output, EventEmitter, Input } from '@angular/core';
 import { UserService } from '@sunbird/core';
 import { environment } from '@sunbird/environment';
+import { ToasterService } from '@sunbird/shared';
+import * as _ from 'lodash-es';
+import { FormBuilder, Validators, FormGroup, FormControl } from '@angular/forms';
 
 @Component({
   selector: 'app-validate-teacher-identifier-popup',
   templateUrl: './validate-teacher-identifier-popup.component.html',
   styleUrls: ['./validate-teacher-identifier-popup.component.scss']
 })
 export class ValidateTeacherIdentifierPopupComponent implements OnInit {
-  /* TODO:  code optimization yet to be done after API integration,
-            msgs to be taken from api response
-  */
+  @Input() userFeedData: {};
   @Output() close = new EventEmitter<any>();
   @ViewChild('createValidateModal') createValidateModal;
-  @ViewChild('extIdInputField') extIdInputField: ElementRef;
-  externalId: string;
+  userDetailsForm: FormGroup;
+  formBuilder: FormBuilder;
   processValidation = false;
   enableSubmitButton: boolean;
-  count = 0;
   showError: boolean;
   extIdVerified: boolean;
   extIdFailed: boolean;
-  navivateToResult =  false;
-  instance: string;
   isOffline = environment.isOffline;
   userId: string;
+  channelData: [];
+  showStateDropdown: boolean;
+  telemetryCdata: Array<{}> = [];
+  telemetryInteractObject: IInteractEventObject;
   constructor(
     public userService: UserService,
-    public resourceService: ResourceService) { }
+    public resourceService: ResourceService,
+    public toasterService: ToasterService) { }
+
   ngOnInit() {
-    this.instance = 'Tamil Nadu';
     this.userId = this.userService.userid;
+    this.processUserFeedData();
+    this.initializeFormField();
   }
 
+  initializeFormField() {
+    if (this.showStateDropdown) {
```

Comment:
this.userDetailsForm = new FormGroup({
        extId: new FormControl('', Validators.required)
      });
if( this.showStateDropdown ) {
userDetailsForm.addControl(name ,control);
}